Dockerfile

```diff
@@ -25,6 +25,30 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
         npm \
         pkg-config
 
+# Used for platform-specific instructions
+ARG TARGETPLATFORM
```

Comment:
Starting a new "thread" on cross-compilation.

Looked into this a bit today, and it's not so simple given the current Dockerfile. `FROM --platform=$BUILDPLATFORM` is only half of the equation - the other half is getting the output to be `TARGETPLATFORM`-native. [tonistiigi/xx](https://github.com/tonistiigi/xx) provides drop-in replacements for build tools like `gcc`, so what's covered by that is not a problem.

The problem is with various package managers not supported by `xx` (e.g. `pip` and `npm`). As far as I can tell, `pip` will install based on the caller's platform. So if using `FROM --platform=$BUILDPLATFORM`, this will not reflect the `TARGETPLATFORM`.

One idea is to have two builder stages:

1. `builder` to compile using `xx` (no emulation, much faster)
2. `builder-emulation` to install from package managers (slower, but dunno how else to do this).

This also isn't ideal because some Python packages have dependencies that don't provide pre-compiled arm64 wheels and thus requires building, which defeats the point of cross-compilation in the first place. We could manually build all these in the non-emulation stage (like 2bc2fa1bf13f12174a667982f40ff63d710fad86), but that makes this Dockerfile very hard to manage since we'd have to account for the whole dependency tree.

`pip install` has a `--platform` option, but it requires either `--only-binary=:all:` or `--no-deps`, none of which makes this useful to us.

This is [a work in progress](https://github.com/nextstrain/docker-base/commit/3aa7a91a4cbd33a47a4bfd040128c20c7b6ffb3b), and I'm still trying to fully grasp these concepts. There's probably a better way to do this.

Reply:
Though it should be possible, it can be tricky to make the whole toolchain work for cross-compilation, esp. across ecosystems like pip/npm/etc.

Reply:
After some extra thought on this, I think the two-builder-stage approach is the best we can get that's still reasonably maintainable. Testing that on a separate branch [victorlin/arm64-cross-compilation](https://github.com/nextstrain/docker-base/tree/victorlin/arm64-cross-compilation), and we can see if the decreased build time (if any) is enough to warrant the extra complexity.

Maybe I'm just bad at searching, but I can't find any examples of other Dockerfiles in the wild that:

1. Use cross-compilation
2. Install a variety of tools via package managers such as `pip` or `npm`

Reply:
Cross-compilation seems to work, and the build time is substantially decreased. I think the changes are complex enough to put in a separate PR: #98